File: Zad4.java
package test;
import java.util.ArrayList;
import java.util.Random;

class Parrot {
    private ArrayList<String> phrases;
    private Random rand;

    public Parrot(String initialPhrase) {
        phrases = new ArrayList<>();
        phrases.add(initialPhrase);
        rand = new Random();
    }

    public void say() {
        System.out.println(phrases.get(rand.nextInt(phrases.size())));
    }

    public void say(int counter) {
        for (int i = 0; i < counter; i++) {
            say();
        }
    }

    public void addPhrase(String newPhrase) {
        phrases.add(newPhrase);
    }
}

public class Main {
    public static void main(String[] args) {
        Parrot papuga = new Parrot("Dzwoni papuga");

        papuga.addPhrase("Bazinga");
        papuga.addPhrase("Test01");
        papuga.addPhrase("Virtual XI");

        papuga.say(3);
    }
}

File: Zad4.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>
using namespace std;

class Parrot {
private:
	vector<string> phrases;

public:
	Parrot(string initialPhrase) {
		srand(static_cast<unsigned int>(time(0)));
		phrases.push_back(initialPhrase);
	}

	void say() const {
		cout << phrases[rand() % phrases.size()] << endl;
	}

	void say(int counter) const {
		for (int i = 0; i < counter; ++i) {
			say();
		}
	}

	void addPhrase(const string& newPhrase) {
		phrases.push_back(newPhrase);
	}
	/*
	Parrot(string initialPhrase) {
        phrase = initialPhrase;
    }
	void setPhrase(string newPhrase) {
		phrase = newPhrase;
	}
	void say() const {
		cout << phrase << endl;
	}
	*/
};

int main() {
	Parrot papuga("Dzwoni papuga");

	papuga.addPhrase("Bazinga");
	papuga.addPhrase("Test01");
	papuga.addPhrase("Virtual XI");

	papuga.say(3);

	return 0;
}